### ISSUE TITLE
[Windows Alerting] Expressions do not work correctly with Windows nodes so need to warn user

### ISSUE THREAD

**Issue by davidnuzik** [rancher/rancher#20122]
Version: Master (v2.3) (5/9/19)

**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
- Create a cluster with windows nodes (1 or more)
- Peg the CPU on the windows nodes
- Enable monitoring
- Create an alert that uses expression to check CPU usage. Set it to alert at say 40%

**Result:**
Alert will not fire (assuming no workloads, etc) because it's not accounting for the windows nodes because we cannot (not possible today to deploy prometheus/monitoring to windows nodes).

Because of this we need to have some kind of warning in the UI explaining that metrics will not be captured for windows nodes (if and only if windows nodes are detected)

**Other details that may be helpful:**
If this is too much work for alpha:
> Because of this we need to have some kind of warning in the UI explaining that metrics will not be captured for windows nodes (if and only if windows nodes are detected)

Then  just showing any warning message regardless if okay.

**Comment by davidnuzik** [comment]
Version: master-head (v2.3) (8/29/19)

I don't seem to be able to see my windows node listed:
`sum(node_load1) by (instance) / count(node_cpu_seconds_total{mode="system"}) by (instance) * 100`

I simply reused the default CPU load example we have with this expression. Only my linux master and linux worker node show up in the graph in the UI. The windows node is not listed.

Is there something wrong with this expression? Do we need to change the recommended default one in the UI?

Screenshot of the default one I'm talking about that I took this expression from:
![image](https://user-images.githubusercontent.com/45179589/63979751-79c9ea80-ca6e-11e9-8d0f-61db94eb9edd.png)

Oddly enough my CPU was above my threshold too and it never fired (even though not counting the windows node).

**Comment by davidnuzik** [comment]
Reopening for review. Not sure if something is wrong with this expression or for some reason is not supported with Windows or something.

**Comment by aiwantaozi** [comment]
Update pre-canned cluster level node alert to support both windows and linux. For cpu and memory, the expression could support both linux and windows. For disk running full, if your cluster has windows node, we will add one more alert for windows disk running full. If we use same disk alert for windows and linux, that expression is inaccuracy for linux, so we separate them.

**Comment by davidnuzik** [comment]
Testing on hold due to https://github.com/rancher/rancher/issues/22615

**Comment by davidnuzik** [comment]
I am still running into the issue from my last comment https://github.com/rancher/rancher/issues/20122#issuecomment-526380456

When I use this CPU expression I'm not getting any metrics shown for my windows node.
`sum(node_load1) by (instance) / count(node_cpu_seconds_total{mode="system"}) by (instance) * 100`

I should see three nodes, 1 linux master, 1 linux worker, 1 windows worker. The windows worker does not show up.
![image](https://user-images.githubusercontent.com/45179589/64374420-af278880-cfd8-11e9-86d0-b0518ea5e3de.png)

**Comment by davidnuzik** [comment]
Version: master-head (v2.3) (9/5/19)

I can confirm the new default node alerts are working correctly. The expression was changed, but still works in an expected manner now. There was an extra alert added for a windows disk check (it must be a separate alert). I made sure the new high cpu load expression (just changed formula) works and the new windows low disk alert works (new alert).

**_Examples and tests:_**

**Node high cpu load expression** was changed but still results in the same output (works the same). Now we can see our windows node with the new expression `sum(node_load1) by (node)  / sum(machine_cpu_cores) by (node) * 100` The old expression was `sum(node_load1) by (instance) / count(node_cpu_seconds_total{mode="system"}) by (instance) * 100` which did not work.

As we can see in the screenshot, my windows node is listed now:
![image](https://user-images.githubusercontent.com/45179589/64379222-48f43300-cfe3-11e9-8a88-90f73234c261.png)

And when I cause high CPU load on the windows node, the alert will fire and send a notification to me (tested with slack).

**High node memory utilization expression** also works as expected. My windows node is listed. 
![image](https://user-images.githubusercontent.com/45179589/64379443-c324b780-cfe3-11e9-9952-5718e31f533d.png)

**Windows node disk is running full within 24 hours** was added. This seems to be set to "Not Null" and the expression actually results in a value (amount of bytes but in a negative number it seems). I'm not quite sure if this is what we want although the alert did not go off unexpectedly for me.

![image](https://user-images.githubusercontent.com/45179589/64380712-6d054380-cfe6-11e9-99e7-ff70e309fd57.png)

I tested this vis `fsutil` you can use it to create files of an arbitrary size. This just tells the filesystem the file is of n size but doesn't actually require writing all the bits to disk to actually take up the space.
Example: `fsutil file createnew file.txt 5000000000` This will create a text file that is 5GB (approx) in size.

While I did this I saw the disk free space decrease, thus changing the number on the graph.
![image](https://user-images.githubusercontent.com/45179589/64386709-b22d7380-cfee-11e9-8f37-c1c8c28ed173.png)

----------

I never quite saw the _Windows node disk is running full within 24 hours_ actually alert @aiwantaozi Am I missing something? Do I need to wait 24 hours? I tried playing with the expression, anyway I could make this a shorter amount of time?

I took up 99% of disk space and never saw it fire within about a 1 hour timeframe.

**Comment by davidnuzik** [comment]
@aiwantaozi could you review the last section around Windows node disk is running full within 24 hours and my question when you are able? I was not certain if this was working correctly and I just needed to wait longer or if there is an actual issue or not.

If I need to wait 24 hours I presume monitoring should also be set up for 24 (or longer) timeframe yes? By default monitoring only keeps data for 12 hours.

**Comment by davidnuzik** [comment]
I am also seeing non-windows node disk alert firing even though node disks shouldn't be nearly full.
![image](https://user-images.githubusercontent.com/45179589/64389357-2ff57d00-cff7-11e9-8bc5-fabb7dcc6456.png)

**Comment by aiwantaozi** [comment]
@davidnuzik For `Windows node disk is running full within 24 hours`, it should send out alerts when there is negative data, could you check the click the link for app rancher-monitoring and go to Prometheus alert page to see whether there are active alerts. Like this, I just update the expression to make it could be triggered, ignore the expression part. 
![image](https://user-images.githubusercontent.com/5936271/64398514-b06bba80-d097-11e9-9193-8ac024dde95e.png)

If there are active alerts, check the alertmanager to see whether the alertmanager receive it
![image](https://user-images.githubusercontent.com/5936271/64398643-26702180-d098-11e9-849f-efd94b96dfa5.png)

The predict_linear is a function from prometheus, it uses the last 6h to predict incoming 24h,  no need to wait for 24h.

**Comment by davidnuzik** [comment]
@aiwantaozi It looks like now there is no metric graph data on latest master-head

![image](https://user-images.githubusercontent.com/45179589/64465523-f0e02e00-d0c1-11e9-8cfb-7137f3f417b3.png)

If I attempt to execute the expression in prometheus it reports "No datapoints found."

**Comment by aiwantaozi** [comment]
Yes, you could see it would alert when the value is `not null`, so it wouldn't have any value, this is an alert for the disk is running full within 24h, also an example to tell the user they can input an expression without threshold.

**Comment by loganhz** [comment]
Could you create another issue for `Not Null` condition? Maybe we can think about how to fix it in 2.3.x Thanks!

**Comment by davidnuzik** [comment]
https://github.com/rancher/rancher/issues/22742
